src/checkout-buttons/strategies/googlepay/googlepay-adyenv2-button-strategy.spec.ts

```diff
@@ -0,0 +1,172 @@
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.createButton).toHaveBeenCalled();
+            });
+
+            it('Validates if strategy has been initialized', async () => {
+                checkoutButtonOptions = getCheckoutButtonOptions();
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.initialize).toHaveBeenCalledTimes(1);
+            });
+
+            it('fails to initialize the strategy if no container id is supplied', () => {
+                checkoutButtonOptions = getCheckoutButtonOptions(Mode.UndefinedContainer);
+
+                expect(() => strategy.initialize(checkoutButtonOptions)).toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if no valid container id is supplied', () => {
+                checkoutButtonOptions = getCheckoutButtonOptions(Mode.InvalidContainer);
+
+                return expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(InvalidArgumentError);
+            });
+        });
+    });
+
+    describe('#deinitialize()', () => {
+        let containerId: string;
+
+        beforeAll(() => {
+            checkoutButtonOptions = getCheckoutButtonOptions(Mode.GooglePayAdyenV2);
+            containerId = checkoutButtonOptions.containerId;
+        });
+
+        it('check if googlepay payment processor deinitialize is called', async () => {
+            await strategy.initialize(checkoutButtonOptions);
+
+            await strategy.deinitialize();
+
+            expect(paymentProcessor.deinitialize).toBeCalled();
+        });
+
+        it('removes the child elements from attached to the containerId', async () => {
+            await strategy.initialize(checkoutButtonOptions);
+
+            const button = document.getElementById(containerId);
+
+            expect(button).toHaveProperty('firstChild', walletButton);
+
+            await strategy.deinitialize();
+
+            expect(button).toHaveProperty('firstChild', null);
+        });
+
+        it('fails silently if strategy was not initialized', () => {
+            expect(async () => await strategy.deinitialize()).not.toThrow();
+        });
+    });
+
+    describe('#handleWalletButtonClick', () => {
+        it('handles wallet button event', async () => {
+            checkoutButtonOptions = getCheckoutButtonOptions(Mode.GooglePayAdyenV2);
+
+            jest.spyOn(paymentProcessor, 'displayWallet').mockReturnValue(Promise.resolve(getGooglePaymentDataMock()));
+            jest.spyOn(paymentProcessor, 'handleSuccess').mockReturnValue(Promise.resolve());
+            jest.spyOn(paymentProcessor, 'updateShippingAddress').mockReturnValue(Promise.resolve());
+
+            await strategy.initialize(checkoutButtonOptions);
+
+            walletButton.click();
+
+            expect(paymentProcessor.initialize).toHaveBeenCalled();
```

Comment:
wouldn't this be called even if `walletButton.click()`  line was removed? what are you exactly expecting to happen after `walletButton.click();` is called?

```js
// ... rest of the test
await strategy.initialize(checkoutButtonOptions);
expect(paymentProcessor.initialize).toHaveBeenCalled()

walletButton.click();
// expect ??
```